File: lib/presentation/widgets/sections/trusted_metrics_section.dart
import 'package:flutter/material.dart';
import '../../../core/constants/app_colors.dart';

class TrustedMetricsSection extends StatelessWidget {
  const TrustedMetricsSection({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final screenSize = MediaQuery.of(context).size;
    final isMobile = screenSize.width < 768;

    return Container(
      color: AppColors.metricsBackground,
      padding: EdgeInsets.symmetric(
        vertical: 72,
        horizontal: isMobile ? 24 : 48,
      ),
      child: Center(
        child: Container(
          constraints: const BoxConstraints(maxWidth: 1200),
          child: Column(
            children: [
              _buildSectionHeader(isMobile),
              SizedBox(height: isMobile ? 40 : 48),
              isMobile
                  ? Column(
                      children: _buildMetricCards(isMobile),
                    )
                  : Row(
                      mainAxisAlignment: MainAxisAlignment.spaceAround,
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: _buildMetricCards(isMobile),
                    ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildSectionHeader(bool isMobile) {
    return Column(
      children: [
        Text(
          'Trusted by Students Worldwide',
          style: TextStyle(
            fontSize: isMobile ? 28 : 36,
            fontWeight: FontWeight.w800,
            color: AppColors.metricsText,
            fontFamily: 'Inter',
          ),
          textAlign: TextAlign.center,
        ),
        SizedBox(height: isMobile ? 12 : 16),
        Text(
          'Join thousands of successful graduate school applicants',
          style: TextStyle(
            fontSize: isMobile ? 16 : 18,
            color: AppColors.metricsSubtext,
            fontFamily: 'Inter',
          ),
          textAlign: TextAlign.center,
        ),
      ],
    );
  }

  List<Widget> _buildMetricCards(bool isMobile) {
    final metrics = [
      {
        'icon': Icons.school,
        'number': '15,000+',
        'label': 'Graduate Programs Reviewed',
      },
      {
        'icon': Icons.emoji_events,
        'number': '3,200+',
        'label': 'Successful Applicants',
      },
      {
        'icon': Icons.location_city,
        'number': '500+',
        'label': 'Universities Covered',
      },
      {
        'icon': Icons.trending_up,
        'number': '98%',
        'label': 'Success Rate',
      },
    ];

    return metrics.asMap().entries.map((entry) {
      final index = entry.key;
      final metric = entry.value;
      
      return Container(
        margin: EdgeInsets.only(
          bottom: isMobile && index < metrics.length - 1 ? 32 : 0,
        ),
        child: _buildMetricCard(
          icon: metric['icon'] as IconData,
          number: metric['number'] as String,
          label: metric['label'] as String,
        ),
      );
    }).toList();
  }

  Widget _buildMetricCard({
    required IconData icon,
    required String number,
    required String label,
  }) {
    return Column(
      children: [
        // Icon with circular background
        Container(
          width: 56,
          height: 56,
          decoration: BoxDecoration(
            color: Colors.white.withOpacity(0.12),
            shape: BoxShape.circle,
          ),
          child: Icon(
            icon,
            size: 28,
            color: AppColors.metricsText,
          ),
        ),
        const SizedBox(height: 16),
        
        // Number
        Text(
          number,
          style: const TextStyle(
            fontSize: 28,
            fontWeight: FontWeight.w800,
            color: AppColors.metricsText,
            fontFamily: 'Inter',
          ),
        ),
        const SizedBox(height: 8),
        
        // Label
        Container(
          constraints: const BoxConstraints(maxWidth: 200),
          child: Text(
            label,
            style: const TextStyle(
              fontSize: 16,
              color: AppColors.metricsSubtext,
              fontFamily: 'Inter',
            ),
            textAlign: TextAlign.center,
          ),
        ),
      ],
    );
  }
}

File: src/components/TrustedMetricsSection.js
import React from 'react';
import { School, Trophy, Building2, TrendingUp } from 'lucide-react';
import { colors, textStyles, spacing, sectionSpacing } from '../theme';

const TrustedMetricsSection = () => {
  const isMobile = window.innerWidth < 768;

  const metrics = [
    {
      icon: School,
      number: '15,000+',
      label: 'Graduate Programs Reviewed',
    },
    {
      icon: Trophy,
      number: '3,200+',
      label: 'Successful Applicants',
    },
    {
      icon: Building2,
      number: '500+',
      label: 'Universities Covered',
    },
    {
      icon: TrendingUp,
      number: '98%',
      label: 'Success Rate',
    },
  ];

  return (
    <section
      style={{
        backgroundColor: colors.metricsBackground,
        padding: `${sectionSpacing[isMobile ? 'mobile' : 'desktop'].small} ${sectionSpacing.horizontal[isMobile ? 'mobile' : 'desktop']}`,
      }}
    >
      <div
        style={{
          maxWidth: '1200px',
          margin: '0 auto',
        }}
      >
        {/* Section Header */}
        <div style={{ textAlign: 'center', marginBottom: isMobile ? spacing[10] : spacing[12] }}>
          <h2
            style={{
              ...textStyles[isMobile ? 'sectionTitleMobile' : 'sectionTitle'],
              color: colors.metricsText,
              marginBottom: isMobile ? spacing[3] : spacing[4],
            }}
          >
            Trusted by Students Worldwide
          </h2>
          <p
            style={{
              ...textStyles.sectionSubtitle,
              color: colors.metricsSubtext,
            }}
          >
            Join thousands of successful graduate school applicants
          </p>
        </div>

        {/* Metrics Grid */}
        <div
          style={{
            display: isMobile ? 'flex' : 'grid',
            flexDirection: isMobile ? 'column' : undefined,
            gridTemplateColumns: isMobile ? undefined : 'repeat(4, 1fr)',
            gap: isMobile ? spacing[8] : spacing[4],
            alignItems: 'center',
            justifyContent: 'space-around',
          }}
        >
          {metrics.map((metric, index) => (
            <MetricCard key={index} metric={metric} />
          ))}
        </div>
      </div>
    </section>
  );
};

const MetricCard = ({ metric }) => {
  const { icon: Icon, number, label } = metric;

  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        textAlign: 'center',
      }}
    >
      {/* Icon with circular background */}
      <div
        style={{
          width: '56px',
          height: '56px',
          backgroundColor: 'rgba(255, 255, 255, 0.12)',
          borderRadius: '50%',
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
          marginBottom: spacing[4],
        }}
      >
        <Icon size={28} color={colors.metricsText} />
      </div>

      {/* Number */}
      <div
        style={{
          ...textStyles.metricNumber,
          color: colors.metricsText,
          marginBottom: spacing[2],
        }}
      >
        {number}
      </div>

      {/* Label */}
      <div
        style={{
          ...textStyles.metricLabel,
          color: colors.metricsSubtext,
          maxWidth: '200px',
        }}
      >
        {label}
      </div>
    </div>
  );
};

export default TrustedMetricsSection;